src/Tests/dotnet.Tests/CommandTests/ToolUpdateGlobalOrToolPathCommandTests.cs

```diff
@@ -284,6 +285,53 @@ public void GivenAnExistedLowerversionWhenReinstallThrowsItRollsBack()
                 .Be(LowerPackageVersion);
         }
 
+        [Fact]
+        public void GivenPackagedShimIsProvidedWhenRunWithPackageIdItCreatesShimUsingPackagedShim()
+        {
+            CreateInstallCommand($"-g {_packageId} --version {LowerPackageVersion}").Execute();
+            _reporter.Lines.Clear();
+
+            var extension = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : string.Empty;
+            var prepackagedShimPath = Path.Combine(Path.GetTempPath(), "SimulatorCommand" + extension);
+            var tokenToIdentifyPackagedShim = "packagedShim";
+            _fileSystem.File.WriteAllText(prepackagedShimPath, tokenToIdentifyPackagedShim);
+
+            var packagedShimsMap = new Dictionary<PackageId, IReadOnlyList<FilePath>>
+            {
+                [_packageId] = new[] {new FilePath(prepackagedShimPath)}
+            };
+
+            string options = $"-g {_packageId}";
+            ParseResult result = Parser.Instance.Parse("dotnet tool update " + options);
+            var command = new ToolUpdateGlobalOrToolPathCommand(
```

Comment:
src/Tests/dotnet.Tests/CommandTests/ToolUpdateGlobalOrToolPathCommandTests.cs(306,31): error CS1729: (NETCORE_ENGINEERING_TELEMETRY=Build) 'ToolUpdateGlobalOrToolPathCommand' does not contain a constructor that takes 5 arguments